### PR TITLE
Adding feedback actions bar component.

### PR DESCRIPTION
**Pull request by cbusmand2l**
image

### CODE REVIEW

File: components/consistent-evaluation-feedback-actions.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+	static get properties() {
+		return {
+			canAddFile: { type: Boolean },
+			canAddLink: { type: Boolean },
+			canAddGoogleDriveLink: { type: Boolean },
+			canAddOneDriveLink: { type: Boolean },
+			canRecordAudio: { type: Boolean },
+			canRecordVideo: { type: Boolean },
+			showAll: { type: Boolean }
```
Comment (cbusmand2l) [inline components/consistent-evaluation-feedback-actions.js]: `showAll` will override all the other `canAdd...` etc. options, and just show all defined buttons.

File: components/consistent-evaluation-feedback-actions.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	constructor() {
+		super();
+
+		this._tooltipBoundary = {
+			left: 20 + 12, // padding-left applied to d2l-consistent-evaluation-feedback-actions + padding-left of d2l-button-icon
+			right: 0
+		};
+	}
+
+	_dispatchButtonClick(name) {
```
Comment (cbusmand2l) [inline components/consistent-evaluation-feedback-actions.js]: This component is the dumb UI-only part. I don't think we have enough information at this time to determine where we're going to handle the actual adding of links/recordings/etc, so for now this is just dispatching the button click events for its eventual parent to handle.

**Comment by cbusmand2l** [comment]
The perceptual diff background colour makes it look like the background fill on this component isn't working (but it's definitely there):
image

File: components/consistent-evaluation-feedback-actions.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		}
+
+		return null;
+	}
+
+	constructor() {
+		super();
+
+		this._tooltipBoundary = {
+			left: 20 + 12, // padding-left applied to d2l-consistent-evaluation-feedback-actions + padding-left of d2l-button-icon
```
Comment (jeroach) [inline components/consistent-evaluation-feedback-actions.js]: I haven't seen this pattern before - did you copy from somewhere or come up with on your own?

Just curious :)

File: components/consistent-evaluation-feedback-actions.js (hunk 2 above)
Comment (jeroach) [reply components/consistent-evaluation-feedback-actions.js]: Sounds reasonable

File: components/consistent-evaluation-feedback-actions.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+					id="${uniqueId}"
+					icon="d2l-tier1:${icon}"
+					text="${text}"
+					@click="${handler}">
+				</d2l-button-icon>
+				<d2l-tooltip
+					for="${uniqueId}"
+					aria-hidden="true"
+					disable-focus-lock
+					.boundary="${this._tooltipBoundary}">${text}</d2l-tooltip>
```
Comment (jeroach) [inline components/consistent-evaluation-feedback-actions.js]: super super nit: should the closing tag be on the following line for consistency?

File: components/consistent-evaluation-feedback-actions.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+			`;
+		}
+	}
+	_renderSubtleButton(icon, textKey, eventKey, visible) {
+		if (visible || this.showAll) {
+			const uniqueId = getUniqueId();
+			const text = this.localize(textKey);
+			const handler = () => this._dispatchButtonClick(eventKey);
+			return html`
+				<d2l-button-subtle
```
Comment (jeroach) [inline components/consistent-evaluation-feedback-actions.js]: Do subtle buttons have tooltips?

File: components/consistent-evaluation-feedback-actions.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+					@click=${handler}>
+				</d2l-button-subtle>
+			`;
+		}
+	}
+
+	render() {
+		return html`
+			<div class="button-container-left">
+				${this._renderIconButton('upload', 'addFile', 'add-file', this.canAddFile)}
```
Comment (jeroach) [inline components/consistent-evaluation-feedback-actions.js]: should these strings be constants?

**Comment by cbusmand2l** [comment]
We'll probably import this from activities.

**Pull request closed without merging**